service/controller/appcatalog/resource/appcatalogentry/create.go

```diff
@@ -293,35 +288,33 @@ func (r *Resource) newAppCatalogEntries(ctx context.Context, cr v1alpha1.AppCata
 	return entryCRs, nil
 }
 
-func parseLatestVersion(entries []entry) (string, error) {
+func sortEntry(entries []entry) ([]entry, error) {
 	if len(entries) == 0 {
-		return "", nil
+		return nil, nil
 	}
 
-	versions := make([]*semver.Version, len(entries))
-
-	for i, entry := range entries {
-		v, err := semver.NewVersion(entry.Version)
+	sort.Slice(entries, func(i, j int) bool {
+		prevRelease, err := entries[i].Version.SetPrerelease("")
 		if err != nil {
-			return "", microerror.Mask(err)
+			return false
+		}
+		nextRelease, err := entries[j].Version.SetPrerelease("")
+		if err != nil {
+			return false
 		}
 
-		versions[i] = v
-	}
-
-	// Sort the versions semantically and return the latest.
-	sort.Sort(semver.Collection(versions))
-	latest := versions[len(versions)-1]
+		if prevRelease.GreaterThan(&nextRelease) {
+			return true
+		}
 
-	return latest.String(), nil
-}
+		if prevRelease.Equal(&nextRelease) {
+			if entries[i].Created.After(entries[j].Created.Time) {
```

Comment:
if two release versions are the same then we need to compare the creation timestamp.